packages/sit-onyx/src/components/OnyxDataGrid/OnyxDataGrid.vue

```diff
@@ -0,0 +1,44 @@
+<script setup lang="ts" generic="TEntry extends DataGridEntry">
+import { ref, toRefs, watch, type Ref } from "vue";
+import {
+  OnyxDataGridRenderer,
+  type DataGridEntry,
+  type DataGridMetadata,
+  type DataGridRendererColumn,
+  type DataGridRendererRow,
+} from "../..";
+import { useTableFeatures } from "./features";
+import { useTableSorting } from "./features/sorting/sorting";
+
+const props = defineProps<{
+  /**
+   * The order of and which columns should be rendered.
+   */
+  columns: (keyof TEntry)[];
+  /**
+   * The data that should be used to fill the table.
+   */
+  data: TEntry[];
+}>();
+
+const withSorting = useTableSorting<TEntry>();
+
+const { watchSources, createRendererRows, createRendererColumns } = useTableFeatures([withSorting]);
```

Comment:
Aren't the features passed as props by the project?

Reply:
Yes, that will come in later PRs